pkg/operator/ceph/cluster/rbd/mirror.go

```diff
@@ -61,61 +62,59 @@ func (r *ReconcileCephRBDMirror) start(cephRBDMirror *cephv1.CephRBDMirror) erro
 
 	logger.Infof("configure rbd-mirroring with %d workers", cephRBDMirror.Spec.Count)
 
-	for i := 0; i < cephRBDMirror.Spec.Count; i++ {
-		daemonID := k8sutil.IndexToName(i)
-		resourceName := fmt.Sprintf("%s-%s", AppName, daemonID)
-		daemonConf := &daemonConfig{
-			DaemonID:     daemonID,
-			ResourceName: resourceName,
-			DataPathMap:  config.NewDatalessDaemonDataPathMap(cephRBDMirror.Namespace, r.cephClusterSpec.DataDirHostPath),
-			ownerRef:     *ref,
-		}
+	daemonID := k8sutil.IndexToName(0)
+	resourceName := fmt.Sprintf("%s-%s", AppName, daemonID)
```

Comment:
If we always know there will be exactly one, how about if we just name the deployment `AppName`? No need for the `-a` suffix. Perhaps it will also make `removeExtraMirrors()` a bit simpler as well.

Reply:
This is intentional, I don't want to break the existing cluster with mirroring configured, if we remove this, we would need to update all the code that manages the cephx users/keys too. For instance, if we want to remove it, we would need to update mirroring peers with the new name and key, so maintaining existing clusters would be difficult and too error-prone. All of sudden, this "simple" transition becomes really complicated. I realize that leaving the `-a` behind is not satisfying but it's a middle ground.

Reply:
Yeah, agreed it's fine to leave the `-a` daemon